homeassistant/components/whois/sensor.py

```diff
@@ -79,18 +120,31 @@ async def async_setup_entry(
         LOGGER.error("Exception %s occurred during WHOIS lookup for %s", ex, domain)
         return
 
-    async_add_entities([WhoisSensor(domain)], True)
+    async_add_entities(
+        [
+            WhoisSensorEntity(
+                domain=domain,
+                description=description,
+            )
+            for description in SENSORS
+        ],
+        update_before_add=True,
+    )
 
 
-class WhoisSensor(SensorEntity):
+class WhoisSensorEntity(SensorEntity):
     """Implementation of a WHOIS sensor."""
 
-    _attr_icon = "mdi:calendar-clock"
-    _attr_native_unit_of_measurement = TIME_DAYS
+    entity_description: WhoisSensorEntityDescription
 
-    def __init__(self, domain: str) -> None:
+    def __init__(self, description: WhoisSensorEntityDescription, domain: str) -> None:
         """Initialize the sensor."""
+        self.entity_description = description
         self._attr_name = domain
+        self._attr_unique_id = f"{domain}_{description.key}"
+        self._attr_device_info = DeviceInfo(
```

Comment:
would this be type of service?

Reply:
Well spotted 👍 

Added the service entry type.